### PR TITLE
pipeline-manager: query parameter `code=false` excludes `program_info`

### PR DESCRIPTION
**Pull request by snkas**

### CODE REVIEW

File: crates/pipeline-manager/src/api/pipeline.rs [hunk 1]
```diff
@@ -48,7 +48,7 @@ pub struct ExtendedPipelineDescrOptionalCode {
     pub program_version: Version,
     pub program_status: ProgramStatus,
     pub program_status_since: DateTime<Utc>,
-    pub program_info: Option<ProgramInfo>,
+    pub program_info: Option<Option<ProgramInfo>>,
```
Comment (gz) [inline crates/pipeline-manager/src/api/pipeline.rs:51]: That's an ugly type, do we need two Options?

**Review by gz — APPROVED** [review]

File: crates/pipeline-manager/src/api/pipeline.rs (hunk 1 above)
Comment (gz) [reply crates/pipeline-manager/src/api/pipeline.rs:51]: can't we set this to None to imply code=false?

Comment (snkas) [reply crates/pipeline-manager/src/api/pipeline.rs:51]: I agree it's not very pretty, in the follow-up PR later after the release it actually makes a difference that it's two Options (namely, None = don't include the field, Some(None) = have it be null). As such for now it seems the best to keep as-is.

Comment (gz) [reply crates/pipeline-manager/src/api/pipeline.rs:51]: seems kinda complicated to understand does the user see this struct through openapi?

**Pull request merged**